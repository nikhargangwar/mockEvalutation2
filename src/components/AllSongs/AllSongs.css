.card-plus-title{
    width: 80%;
    margin-top: 0px;
    margin: 0 auto;
}

.card-space-title{
    display: flex;
    justify-content: space-between;
    align-items: center;
    margin-bottom: 20px;
}

.title-text{
    
    font-size: 50px;
    font-weight: bolder;
}
.toggle-button button{
   background-color: transparent;
   border: none;
}

.genre-cards{
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;
}

.card-space{
    display: flex;
    flex-wrap: wrap;
    justify-content: space-around;
    gap:30px;
}