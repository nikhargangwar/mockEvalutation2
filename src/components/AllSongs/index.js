/* eslint-disable  */
import React from 'react';
import Card from '../Card';
import './AllSongs.css'
import genreIcon from '../../assets/icon-genre.svg'
function AllSongs({allSongsData,setToggleClicked}) {
 const toggleHandler=()=>{
  setToggleClicked(true)
 }
  return (
    <div className="card-plus-title">
      <div className="card-space-title">
        <div className="title-text">
          all songs 
        </div>
        <div className="toggle-button">
          <button onClick={toggleHandler}><img src={genreIcon} alt="toggle" /></button>
        </div>
      </div>
      <div className="card-space">
       
        {
            // console.log(allSongsData.data)
            allSongsData.data.map((songdata,id) => 
            <Card 
            id = {id}
            key={songdata.id}
            songdata={songdata} 
            />)
        //   <Card />
        }
      </div>
    </div>
  );
}

export default AllSongs;
