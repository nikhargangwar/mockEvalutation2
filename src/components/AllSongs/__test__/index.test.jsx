import React from 'react';
import {
  render, screen, fireEvent, waitFor,
} from '@testing-library/react';
import mockSongData from '../../../mocks/songData';
import AllSongs from '..';

jest.mock('react-router-dom', () => ({
  useNavigate: jest.fn(),
}));

describe('AllSongs', () => {
  it('should render when mounted', () => {
    const { container } = render(<AllSongs
      allSongsData={mockSongData}
      setToggleClicked={false}
    />);
    expect(container).toMatchSnapshot();
  });

  // it('should render cards when mounted', async () => {
  //   render(<AllSongs
  //     allSongsData={mockSongData}
  //     setToggleClicked={false}
  //   />);
  //   await waitFor(() => {
  //     expect(screen.getAllByTestId('testCard')).toHaveLength(2);
  //   });
  // });

  //   it('should have sync button when mounted', () => {
  //     render(<BrowserRouter><SyncButton /></BrowserRouter>);
  //     const syncButton = screen.getByTestId('sync-button');
  //     expect(syncButton).toBeTruthy();
  //   });
  //   it('should have sync button when mounted', () => {
  //     render(<BrowserRouter><SyncButton /></BrowserRouter>);
  //     const syncButton = screen.getByTestId('sync-button');
  //     expect(syncButton).toBeTruthy();
  //   });
});
