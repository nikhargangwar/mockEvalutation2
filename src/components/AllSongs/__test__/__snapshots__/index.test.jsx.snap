// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllSongs should render when mounted 1`] = `
<div>
  <div
    class="card-plus-title"
  >
    <div
      class="card-space-title"
    >
      <div
        class="title-text"
      >
        all songs
      </div>
      <div
        class="toggle-button"
      >
        <button>
          <img
            alt="toggle"
            src="icon-genre.svg"
          />
        </button>
      </div>
    </div>
    <div
      class="card-space"
    >
      <div
        class="grey-bakground"
        data-testid="testCard"
      >
        <div
          class="card-image-wrapper"
        >
          <div
            class="card-image"
          >
            <img
              alt="title"
              src="https://i.scdn.co/image/ab67616d0000b27386a8ab515de4b7aef28cd631"
            />
          </div>
        </div>
        <div
          class="card-details"
        >
          <div
            class="card-text"
          >
            <div
              class="card-heading"
            >
              Lost
            </div>
            <div
              class="card-subheading"
            >
              Maroon 5
            </div>
          </div>
          <div
            class="liked-icon"
          >
            <h3>
              <button
                type="button"
              >
                0
              </button>
            </h3>
            <img
              alt="hearticon"
              src="heart-gray.svg"
            />
          </div>
        </div>
      </div>
      <div
        class="dark-bakground"
        data-testid="testCard"
      >
        <div
          class="card-image-wrapper"
        >
          <div
            class="card-image"
          >
            <img
              alt="title"
              src="https://i.scdn.co/image/ab67616d0000b273442b53773d50e1b5369bb16c"
            />
          </div>
        </div>
        <div
          class="card-details"
        >
          <div
            class="card-text"
          >
            <div
              class="card-heading"
            >
              Maps
            </div>
            <div
              class="card-subheading"
            >
              Maroon 5
            </div>
          </div>
          <div
            class="liked-icon"
          >
            <h3>
              <button
                type="button"
              >
                0
              </button>
            </h3>
            <img
              alt="hearticon"
              src="heart-gray.svg"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
